mm2src/lp_ordermatch/lp_bot.rs

```diff
@@ -254,6 +255,23 @@ impl TradingBotContext {
             _ => {},
         }
     }
+
+    async fn on_ctx_stop(&self, ctx: &MmArc) {
+        info!("on_ctx_stop event received");
+        let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
```

Comment:
It may turn out that `MmCtx::simple_market_maker_bot_ctx` is destroyed when `TradingBotContext::on_ctx_stop` is called.
How about returning from the function if we cannot get an instance of `TradingBotContext`?

Reply:
> It may turn out that `MmCtx::simple_market_maker_bot_ctx` is destroyed when `TradingBotContext::on_ctx_stop` is called. How about returning from the function if we cannot get an instance of `TradingBotContext`?

I think it's not possible if we dispatch the event just before we call `ctx.stop()`, this way we ensure all the ctx members live enough to receive the event and shutdown gracefully. After the event is dispatched we call `ctx.stop()`